libs/debugging/include/hpx/debugging/print.hpp

```diff
@@ -23,9 +25,17 @@
 #include <vector>
 
 #if defined(__linux) || defined(linux) || defined(__linux__)
-#include <linux/unistd.h>
 #include <sys/mman.h>
-#define DEBUGGING_PRINT_LINUX
+#include <unistd.h>
+#elif defined(__APPLE__)
+#include <crt_externs.h>
+#include <unistd.h>
+#define environ (*_NSGetEnviron())
+#elif defined(HPX_WINDOWS)
+#include <winsock2.h>
```

Comment:
Why do we need to `#include <winsock2.h>` now?

Reply:
The symbol `char** _environ` is defined in `#include <stdlib.h>` if this is what you were looking for.

Reply:
The CI windows build failed, so I replaced winsock with winsock2 and now it works. I haven't used windows in a while and don't really remember the details - the call to `gethostname` was the problem. It seems to be compiling without errors or warnings on all our test machines now.